### PR TITLE
[hardware, recipe, trainer] feat: add Ascend A3 SFT training script

### PR DESCRIPTION
**Pull request by Simba2017**
### What does this PR do?
This PR adds a new SFT (Supervised Fine-Tuning) training script to support large language model training on Ascend A3 hardware (single node with 16 NPUs). The script is optimized for the Ascend A3 architecture, ensuring full compatibility with the platform's unique hardware features and improving training stability & efficiency for LLMs.
No related GitHub issues or PRs exist prior to this submission, as this is the first support for Ascend A3 in SFT training scripts.

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `fully_async`, `one_step_off`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test
pre-commit run --files examples/sft/gsm8k/run_qwen3_06b_ascend_a3_fsdp.sh

Passed.

This script is validated on Ascend A3 single-node 16 NPUs with real LLM training experiments:
- Successfully runs SFT training with 16 NPUs in distributed mode
- Stable training loss convergence
- No hardware compatibility errors / NPU OOM issues
- Optimized batch size & distributed config for 16-card A3 platform

Test Environment
- Hardware: Ascend A3 Single Node 16 NPUs
- Docker Image: quay.io/ascend/verl:verl-8.5.0-a3-ubuntu22.04-py3.11-v0.7.1
- CANN Version: 8.5.0
- Python: 3.11

```python
bash examples/sft/run_qwen3_06b_ascend_a3_fsdp.sh ./verl_output
```

### Design & Code Changes

- Add example script: `examples/sft/gsm8k/run_qwen3_06b_ascend_a3_fsdp\.sh`

- Adapt Ascend A3 16\-NPU distributed config

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/verl-project/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

**Comment by wuxibin89** [comment]
We're not going to maintain script for every model.
